src/commcare_cloud/commands/ansible/helpers.py

```diff
@@ -35,27 +35,38 @@ def _build_env(self, args):
         return env
 
 
-def get_common_ssh_args(environment, use_pem=False):
+def get_common_ssh_args(cmd_parts, environment, use_factory_auth=False):
     strict_host_key_checking = environment.public_vars.get('commcare_cloud_strict_host_key_checking', True)
 
     common_ssh_args = []
-    if use_pem:
-        pem = environment.public_vars.get('commcare_cloud_pem', None)
-        if pem:
-            common_ssh_args.extend(['-i', pem])
+
+    if use_factory_auth:
+        cmd_parts, common_ssh_args = add_factory_auth_cmd(environment, common_ssh_args, cmd_parts)
     if not strict_host_key_checking:
         common_ssh_args.append('-o StrictHostKeyChecking=no')
 
     known_hosts_filepath = environment.paths.known_hosts
     if os.path.exists(known_hosts_filepath):
         common_ssh_args.append('-o=UserKnownHostsFile={}'.format(known_hosts_filepath))
 
-    cmd_parts = tuple()
     if common_ssh_args:
         cmd_parts += ('--ssh-common-args', ' '.join(shlex_quote(arg) for arg in common_ssh_args))
     return cmd_parts
 
 
+def add_factory_auth_cmd(environment, common_ssh_args, cmd_parts):
+    root_user = environment.public_vars.get('commcare_cloud_root_user', 'root')
+    if '-u' not in cmd_parts:
+        cmd_parts += ('-u', root_user)
+    if not environment.public_vars.get('commcare_cloud_pem'):
+        cmd_parts += ('--ask-pass',)
+    else:
+        pem = environment.public_vars.get('commcare_cloud_pem', None)
```

Comment:
This expression and the one in the line above one that looks like it are equivalent (i.e. `None` is the default value of the second argument), I think it'd be clearer to read if you pulled this line out above the if/else ("`pem = environment.public_vars.get('commcare_cloud_pem')`") and referenced it in the condition as well.

Reply:
Here: https://github.com/dimagi/commcare-cloud/pull/1702/commits/a33437069826eefc41e9073c51aec4c9cde767d0